From ce0f976bb88b5828373b1579cc2a0d95a8e9f664 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Mon, 28 Apr 2014 11:05:41 +0200
Subject: [PATCH] Do not warn on tainted test without PERL5LIB enviroment
 variable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fix is ported from upstream commit:

commit 0a9d31df86597d1480df81c97b688f90d4fe63ba
Author: Leon Timmermans <fawaka@gmail.com>
Date:   Sun Jul 28 21:05:57 2013 +0200

    Added use warnings to all modules

<https://rt.cpan.org/Public/Bug/Display.html?id=85106>

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 lib/TAP/Parser/SourceHandler/Perl.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/TAP/Parser/SourceHandler/Perl.pm b/lib/TAP/Parser/SourceHandler/Perl.pm
index 9257f6a..46ee1fc 100644
--- a/lib/TAP/Parser/SourceHandler/Perl.pm
+++ b/lib/TAP/Parser/SourceHandler/Perl.pm
@@ -167,7 +167,7 @@ sub _mangle_switches {
     # PERL5LIB as -I switches and place PERL5OPT on the command line
     # in order that it be seen.
     if ( $class->_has_taint_switch($switches) ) {
-        my @perl5lib = split /$Config{path_sep}/, $ENV{PERL5LIB};
+        my @perl5lib = defined $ENV{PERL5LIB} ? split /$Config{path_sep}/, $ENV{PERL5LIB} : ();
         return (
             $libs,
             [   @{$switches},
-- 
1.9.0

